Content.Server/Psionics/PsionicsSystem.cs

```diff
@@ -76,13 +72,16 @@ private void OnMeleeHit(EntityUid uid, AntiPsionicWeaponComponent component, Mel
                     return;
                 }
 
-                if (component.Punish && HasComp<PotentialPsionicComponent>(entity) && !HasComp<PsionicComponent>(entity) && _random.Prob(0.5f))
-                    _electrocutionSystem.TryDoElectrocution(args.User, null, 20, TimeSpan.FromSeconds(5), false);
+                if (component.Punish && !HasComp<PsionicComponent>(entity) && _random.Prob(component.PunishChances))
+                    _electrocutionSystem.TryDoElectrocution(args.User, null, component.PunishSelfDamage, TimeSpan.FromSeconds(component.PunishStunDuration), false);
             }
         }
 
-        private void OnInit(EntityUid uid, PsionicComponent component, ComponentInit args)
+        private void OnInit(EntityUid uid, PsionicComponent component, ComponentStartup args)
         {
+            component.AmplificationSources.Add("Baseline Amplification", _random.NextFloat(component.BaselineAmplification.Item1, component.BaselineAmplification.Item2));
+            component.DampeningSources.Add("Baseline Dampening", _random.NextFloat(component.BaselineDampening.Item1, component.BaselineDampening.Item2));
```

Comment:
Since these are Keys, they should 100% be declared as public constants at the top of the file.